services/announcement_service.py

```diff
@@ -0,0 +1,38 @@
+from typing import Optional, List, Type
+from sqlalchemy.orm import Session
+from models.announcement import Announcement
+from repository.announcement_repository import AnnouncementRepository
+from schemas.announcement_schema import AnnouncementCreate, AnnouncementUpdate, AnnouncementResponse
+
+
+class AnnouncementService:
+
+    def __init__(self, session: Session):
+        self.repository = AnnouncementRepository(session)
+
+    def create_announcement(self, announcement: AnnouncementCreate) -> AnnouncementResponse:
+        # check if announcement already exists
+        if self.repository.get_announcement_by_title(announcement.title):
+            raise ValueError("Announcement with this title already exists.")
+
+        # Check if announcement date is before announcement deadline
+        if announcement.announcement_date >= announcement.announcement_deadline:
+            raise ValueError("Announcement date must be before announcement deadline.")
+
+        return AnnouncementResponse.from_orm(self.repository.add_announcement(announcement))
+
+    def get_all_announcements(self) -> list[AnnouncementCreate]:
+        announcements = self.repository.get_announcements()
+        return [AnnouncementCreate.from_orm(announcement) for announcement in announcements]
+
+    def get_announcement_by_id(self, announcement_id: int) -> Optional[AnnouncementCreate]:
+        return AnnouncementCreate.from_orm(self.repository.get_announcement_by_id(announcement_id))
```

Comment:
`from_orm` is deprecated, use:
set `model_config['from_attributes']=True` and use `model_validate` instead.